lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/NumberedSplit.java

```diff
@@ -62,6 +62,24 @@ public String toString() {
   }
 
   @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    NumberedSplit<?> that = (NumberedSplit<?>) o;
+    return index == that.index;
```

Comment:
Then, okay. This PR passed on Mac/Windows and Yarn(on Docker). As you said, current implementation (including yours) looks enough and correct for now. If there is any problem in the future, we can fix at that time. I'll merge this. Thank you, @tcNickolas .